### PR TITLE
Handle missing sumw2 in to_hist

### PR DESCRIPTION
**Pull request by anpicci**
## Summary
- update `to_hist` to seed placeholder arrays so missing entries no longer cause index errors, and guard downstream histogram creation
- add a regression test covering `to_hist` with a missing sumw2 array and enforcing zero variances when `zero_wgts` is requested

## Testing
- `pytest tests/test_datacard_tools.py`

### CODE REVIEW

File: topeft/modules/datacard_tools.py [hunk 1]
```diff
@@ -34,14 +34,22 @@ def to_hist(arr,name,zero_wgts=False):
...
+
+    for i in range(2):
+        if clipped[i] is None:
+            clipped[i] = np.zeros_like(reference)
 
-    nbins = len(clipped[0])
+    nbins = len(reference)
     h = hist.Hist(hist.axis.Regular(nbins,0,nbins,name=name),storage=bh.storage.Weight())
     if zero_wgts:
         h[...] = np.stack([clipped[0],np.zeros_like(clipped[0])],axis=-1) # Set the bin errors all to 0
```
[the remark this answers is not in the document]
Comment (anpicci) [reply topeft/modules/datacard_tools.py:69]: @codex address that feedback

**Comment by anpicci** [comment]
@codex review

File: topeft/modules/datacard_tools.py [hunk 2]
```diff
@@ -34,14 +34,30 @@ def to_hist(arr,name,zero_wgts=False):
...
+        raise ValueError("Input histogram array contains no data")
+
+    if missing_sumw2 and not zero_wgts:
+        raise ValueError(
+            "Input histogram array is missing sumw2 data but zero_wgts is False"
+        )
+
+    for i in range(2):
+        if clipped[i] is None:
+            clipped[i] = np.zeros_like(reference)
```
[the remark this answers is not in the document]
Comment (anpicci) [reply topeft/modules/datacard_tools.py:64]: @codex address that feedback

**Comment by anpicci** [comment]
@codex review
